src/spikeinterface/postprocessing/correlograms.py

```diff
@@ -18,34 +18,48 @@ class ComputeCorrelograms(AnalyzerExtension):
     """
     Compute auto and cross correlograms.
 
+    In the extracellular electrophysiology context, a correlogram
+    is a visualisation of the results of a cross-correlation
+    between two spike trains. The cross-correlation slides one spike train
+    along another sample-by-sample, taking the correlation at each 'lag'. This results
+    in a plot with 'lag' (i.e. time offset) on the x-axis and 'correlation'
+    (i.e. how similar to two spike trains are) on the y-axis. In this
+    implementation, the y-axis result is the 'counts' of spike matches per
+    time bin (rather than a computer correlation or covariance).
+
+    In the present implementation, a 'window' around spikes is first
+    specified. For example, if a window of 100 ms is taken, we will
+    take the correlation at lags from -50 ms to +50 ms around the spike peak.
+    In theory, we can have as many lags as we have samples. Often, this
+    visualisation is too high resolution and instead the lags are binned
+    (e.g. -50 to -45 ms, ..., -5 to 0 ms, 0 to 5 ms, ...., 45 to 50 ms).
+    When using counts as output, binning the lags involves adding up all counts across
+    a range of lags.
```

Comment:
We should discuss this. style:

I think numpydoc recommends 1-2 sentence quick explanation here then go straight to parameters because docstring renderers shouldn't push all the parameter stuff down. This type of info would go in a Notes section at the end for people that want to go deep into this type of information.

I like the style, but depending on what others (@chrishalcrow @h-mayorquin ) think maybe we use a different style with the information at top instead of at bottom?

Reply:
Hello, thanks Zach I hadn't thought about this. I would vote to follow numpy (but I always say this). They say the extended summary should be "A few sentences giving an extended description. This section should be used to clarify functionality, not to discuss implementation detail or background theory, which should rather be explored in the Notes section"

from https://numpydoc.readthedocs.io/en/latest/format.html#extended-summary, cited by https://numpy.org/doc/1.20/docs/howto_document.html. 

Then most of this (very nice!) explanation would go into a Notes section (and would be great to have something like these descriptions in the docs).

Happy to hear arguments against!

Reply:
Nice! I agree this sounds good, I do feel uncomfortable with such long docstrings. It's useful information to have but definitely clutters. I think Notes is a good place for it!